plugins/modules/ecs_task.py

```diff
@@ -349,23 +349,23 @@ def main():
 
     # Validate Inputs
     if module.params['operation'] == 'run':
-        if 'task_definition' not in module.params and module.params['task_definition'] is None:
+        if module.params['task_definition'] is None:
             module.fail_json(msg="To run a task, a task_definition must be specified")
```

Comment:
There is a better way to handle this: on line 348 there's a 'required_if'.  It should be possible to do:
```
required_if=[
    ('launch_type', 'FARGATE', ['network_configuration']),
    ('operation', 'start', ['task_definition', 'container_instances']),
    ('operation', 'stop', ['task_definition', 'task']),
]

This then lets the main Ansible code handle it with a consistent error message.

Reply:
Great feedback.

I agree it is better to have the main Ansible code handle the parameter validation with a consistent error message. I've made the requested changes.

I've added the some entries to the operation description and also required the `cluster` parameter because AFAIK it is required anytime the ecs_task module is used.